### DISCUSSION TITLE
Suggest adding to toolbar "Sort By", or even adaptive sort by.

### DISCUSSION THREAD

**Discussion by taccess0** [nomacs/nomacs#1036] [Ideas]
Want fast changing of sort by filename, date modified, size, etc. You know what I mean.

Simple: Add a menu icon to the default toolbar for quick access. Done!

Automated: In old days Windows Photo Viewer, it adapt the sort by according to the folder you're opeing the file. I don't know if, for nomacs, that's even do-able or too complicated to achieve.

**Comment by leejuyuu** [comment]
Do you mean this?
![ps_20240316235530](https://github.com/nomacs/nomacs/assets/46868116/d5abc209-bc67-412d-a180-01f474e8b008)

**Reply by taccess0** [reply]
Yes! Where do you get that? I just downloaded the latest beta, looked a bit in the menus, and still can't find it. And what about the adaptive sorting I talked about?

**Comment by scrubbbbs** [comment]
The screenshot is the thumbnail view toolbar, or you can right-click on the image and use the context menu.

**Reply by taccess0** [reply]
Found it! So it has to be in the folder mode to have that toolbar shown. I also have the file explorer (E) enabled and clicking on a folder shows that toolbar as well. Thanks! But it would be better if it wasn't a must to be in folder mode.
